File: leetcode-submissions/1455_Check_If_a_Word_Occurs_As_a_Prefix_of_Any_Word_in_a_Sentence.cpp
class Solution {
public:
    bool is_found(string word,string prefix){
        string temp="";
        for(char ch:word){
            if(temp==prefix){
                return true ;
            }
            temp+=ch;
        }
        return false;
    }
    int isPrefixOfWord(string sentence, string searchWord) {
        int index=1;
        //time complexity =o(n)
        stringstream ss(sentence);
        string word;
        while(ss>>word){
            if(is_found(word,searchWord)){
                return index;
            }
            index++;
        }
        return -1;
    }
};

File: leetcode-submissions/1455_Check_If_a_Word_Occurs_As_a_Prefix_of_Any_Word_in_a_Sentence.ts
function isPrefixOfWord(sentence: string, searchWord: string): number {
    const arr:string[]=sentence.split(" ");
    for(let i:number =0;i<arr.length;i++){
        if(is_found(arr[i],searchWord)){
            return i+1;
        }

    }
    return -1;
};
function is_found(word:string , prefix:string):boolean{
    let temp:string ="";
    for(const ch of word){
        temp+=ch;
        if(temp===prefix){
            return true ;

        }
    }
    if(temp==prefix){
        return ;
    }
    return false;
}